arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedDeltaEncodedValuesReader.java

```diff
@@ -0,0 +1,276 @@
+ *     Parquet format encodings: DELTA_BINARY_PACKED</a>
+ */
+public class VectorizedDeltaEncodedValuesReader extends ValuesReader
+    implements VectorizedValuesReader {
+
+  // header data
+  private int blockSizeInValues;
+  private int miniBlocksPerBlock;
+  private int totalValueCount;
+  private long firstValue;
+
+  private int miniBlockSizeInValues;
+
+  // values read by the caller
+  private int valuesRead = 0;
+
+  // variables to keep state of the current block and miniblock
+  private long lastValueRead; // needed to compute the next value
+  private long minDeltaInCurrentBlock; // needed to compute the next value
+  // currentMiniBlock keeps track of the mini block within the current block that
+  // we read and decoded most recently. Only used as an index into
+  // bitWidths array
+  private int currentMiniBlock = 0;
+  private int[] bitWidths; // bit widths for each miniBlock in the current block
+  private int remainingInBlock = 0; // values in current block still to be read
+  private int remainingInMiniBlock = 0; // values in current mini block still to be read
+  private long[] unpackedValuesBuffer;
+
+  private ByteBufferInputStream inputStream;
+
+  // temporary buffers used by readInteger and readLong
+  private int intVal;
+  private long longVal;
+
+  @Override
+  public void initFromPage(int valueCount, ByteBufferInputStream in) throws IOException {
+    Preconditions.checkArgument(
+        valueCount >= 1, "Page must have at least one value, but it has " + valueCount);
+    this.inputStream = in;
+    // Read the header
+    this.blockSizeInValues = BytesUtils.readUnsignedVarInt(this.inputStream);
+    this.miniBlocksPerBlock = BytesUtils.readUnsignedVarInt(this.inputStream);
+    double miniSize = (double) blockSizeInValues / miniBlocksPerBlock;
+    Preconditions.checkArgument(
+        miniSize % 8 == 0, "miniBlockSize must be multiple of 8, but it's " + miniSize);
+    this.miniBlockSizeInValues = (int) miniSize;
+    // True value count. May be less than valueCount because of nulls
+    this.totalValueCount = BytesUtils.readUnsignedVarInt(this.inputStream);
+    this.bitWidths = new int[miniBlocksPerBlock];
+    this.unpackedValuesBuffer = new long[miniBlockSizeInValues];
+    // read the first value
+    firstValue = BytesUtils.readZigZagVarLong(this.inputStream);
+  }
+
+  @Override
+  public byte readByte() {
+    throw new UnsupportedOperationException("readByte is not supported");
+  }
+
+  @Override
+  public short readShort() {
+    throw new UnsupportedOperationException("readShort is not supported");
+  }
+
+  @Override
+  public int readInteger() {
+    readValues(1, null, 0, INT_SIZE, (f, i, v) -> intVal = (int) v);
+    return intVal;
+  }
+
+  @Override
+  public long readLong() {
+    readValues(1, null, 0, LONG_SIZE, (f, i, v) -> longVal = v);
+    return longVal;
+  }
+
+  @Override
+  public void skip() {
+    throw new UnsupportedOperationException("skip is not supported");
```

Comment:
I think maybe we should leave some comments on these unsupported operations as to why Iceberg's reader doesn't use these (Ie we don't do skipping) we (Binary isn't supported in parquet for delta encoding) just to have the context here